api/src/main/java/uk/gov/dvsa/mot/vehicle/hgv/HgvVehicleProvider.java

```diff
@@ -0,0 +1,143 @@
+package uk.gov.dvsa.mot.vehicle.hgv;
+
+import com.google.inject.Inject;
+
+import org.apache.log4j.Logger;
+import org.glassfish.jersey.apache.connector.ApacheConnectorProvider;
+import org.glassfish.jersey.client.ClientConfig;
+import org.glassfish.jersey.client.ClientProperties;
+import org.glassfish.jersey.client.JerseyClientBuilder;
+
+import uk.gov.dvsa.mot.vehicle.hgv.model.TestHistory;
+import uk.gov.dvsa.mot.vehicle.hgv.model.Vehicle;
+import uk.gov.dvsa.mot.vehicle.hgv.response.ResponseTestHistory;
+import uk.gov.dvsa.mot.vehicle.hgv.response.ResponseVehicle;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.Response;
+
+public class HgvVehicleProvider {
+    private static final Logger logger = Logger.getLogger(HgvVehicleProvider.class);
+
+    private HgvConfiguration configuration;
+
+    @Inject
+    public void setHgvConfiguration(HgvConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    public Vehicle getVehicle(String registration) throws Exception {
+        logger.trace("Entering HgvVehicleProvider getVehicle()");
+
+        if (registration == null || registration.isEmpty()) {
```

Comment:
I think we can guarantee registration is not empty much earlier and clean this out.